hubblestack/extmods/returners/splunk_pulsar_return.py

```diff
@@ -315,203 +312,3 @@ def _dedupList(l):
         if x not in l[i + 1:]:
             deduped.append(x)
     return deduped
-
-
-def _get_options():
-    if __salt__['grains.get']('hubblestack:returner:splunk'):
-        splunk_opts = []
-        returner_opts = __salt__['grains.get']('hubblestack:returner:splunk')
-        if not isinstance(returner_opts, list):
-            returner_opts = [returner_opts]
-        for opt in returner_opts:
-            processed = {}
-            processed['token'] = opt.get('token')
-            processed['indexer'] = opt.get('indexer')
-            processed['port'] = str(opt.get('port', '8088'))
-            processed['index'] = opt.get('index')
-            processed['custom_fields'] = opt.get('custom_fields', [])
-            processed['sourcetype'] = opt.get('sourcetype_pulsar', 'hubble_fim')
-            processed['http_event_server_ssl'] = opt.get('hec_ssl', True)
-            processed['proxy'] = opt.get('proxy', {})
-            processed['timeout'] = opt.get('timeout', 9.05)
-            processed['http_event_collector_ssl_verify'] = opt.get('http_event_collector_ssl_verify', True)
-            if 'fallback_indexer' in opt and __grains__.get('ip_gw', None) is False:
-                processed['indexer'] = opt['fallback_indexer']
-            splunk_opts.append(processed)
-        return splunk_opts
-    elif __salt__['config.get']('hubblestack:returner:splunk'):
-        splunk_opts = []
-        returner_opts = __salt__['config.get']('hubblestack:returner:splunk')
-        if not isinstance(returner_opts, list):
-            returner_opts = [returner_opts]
-        for opt in returner_opts:
-            processed = {}
-            processed['token'] = opt.get('token')
-            processed['indexer'] = opt.get('indexer')
-            processed['port'] = str(opt.get('port', '8088'))
-            processed['index'] = opt.get('index')
-            processed['custom_fields'] = opt.get('custom_fields', [])
-            processed['sourcetype'] = opt.get('sourcetype_pulsar', 'hubble_fim')
```

Comment:
It's easy to miss, but this line is different in each of the returners. The result is that all data is being returned under `hubble_log`, rather than `hubble_osquery` or `hubble_audit`, or what have you.

Can you find a way to parameterize this piece of config? I figure you add an optional parameter to `get_splunk_options()` so that a returner can define how you would look up its sourcetype.

Reply:
ugh. this is harder than I thought.

Reply:
wait, maybe not... I put all my new updates in the stash ... testing. I think I handled this but don't recall how it works...

Reply:
yeah. this is easy again. nevermind the several hours I wasted making this difficult.

Reply:
Tested and working.